diff --git a/dist/index.cjs b/dist/index.cjs
index fe08a46803a2e3ef62b0a447c5204095bdac5856..1f64e8b7739253238579c8f5d302104575e48813 100644
--- a/dist/index.cjs
+++ b/dist/index.cjs
@@ -607,7 +607,7 @@ function trainCase(str, opts) {
 }
 var titleCaseExceptions = /^(a|an|and|as|at|but|by|for|if|in|is|nor|of|on|or|the|to|with)$/i;
 function titleCase(str, opts) {
-  return (Array.isArray(str) ? str : splitByCase(str)).filter(Boolean).map(
+  return (Array.isArray(str) ? str : splitByCase(str)).filter(Boolean).map(p => p.trim()).map(
     (p) => titleCaseExceptions.test(p) ? p.toLowerCase() : upperFirst(opts?.normalize ? p.toLowerCase() : p)
   ).join(" ");
 }
diff --git a/dist/index.js b/dist/index.js
index 2e53d0e55b9c028eea628e49af55cbf81e74ed1c..9ccd875b04a0e049716e0f79fe24890d3727a733 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -509,7 +509,7 @@ function trainCase(str, opts) {
 }
 var titleCaseExceptions = /^(a|an|and|as|at|but|by|for|if|in|is|nor|of|on|or|the|to|with)$/i;
 function titleCase(str, opts) {
-  return (Array.isArray(str) ? str : splitByCase(str)).filter(Boolean).map(
+  return (Array.isArray(str) ? str : splitByCase(str)).filter(Boolean).map(p => p.trim()).map(
     (p) => titleCaseExceptions.test(p) ? p.toLowerCase() : upperFirst(opts?.normalize ? p.toLowerCase() : p)
   ).join(" ");
 }
